src/TimeZones.jl

```diff
@@ -76,6 +77,25 @@ function TimeZone(str::AbstractString)
     end
 end
 
+
+"""
+    @tz_str str -> TimeZone
+
+Constructs a `TimeZone` subtype based upon the string `str`.  See docstring of `TimeZone`
+for reference.
+
+```julia
+julia> TimeZone("Africa/Nairobi")
+Africa/Nairobi (UTC+3)
+
+julia> tz"Africa/Nairobi"
+Africa/Nairobi (UTC+3)
+```
+"""
+macro tz_str(str)
+    return :( TimeZone($str) )
```

Comment:
The docstring of `macro` itself puts spaces inside the quoting parentheses, but I will change this.